communication/src/TlsTCPCommunication.cpp

```diff
@@ -23,7 +23,10 @@ TlsTCPCommunication::TlsTCPCommunication(const TlsTcpConfig &config) : config_(c
   // Runners can actually support multiple principals. The Communication interface does not though. Currently we are
   // focused on backwards compatibility, and will use the future runner functionality to replace client thread pools.
   const auto configs = std::vector<TlsTcpConfig>{config};
-  runner_.reset(new tls::Runner(configs, NUM_THREADS));
+  if ((uint32_t)config.selfId > (uint32_t)config.maxServerId)
```

Comment:
You only need to cast one of these values. Also, you should really use `static_cast<uint32>` rather than c-style casts.

Reply:
Done, thanks :)